### PR TITLE
NIFI-10488 Fix NullPointerException on updating empty ParameterContext property with NiFi Toolkit

### PR DESCRIPTION
**Pull request by nandorsoma**
# Summary

[NIFI-10488](https://issues.apache.org/jira/browse/NIFI-10488)

CLI throws an error when we are trying to set an empty property:
```
ERROR: Error executing command 'set-param' : null 
```
Command to test with:
```
./cli.sh nifi set-param -u http://localhost:8080/ -pcid 32d572a1-0183-1000-a811-ab916612b4af -pn emptyParam -pv newvalue
```

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/params/SetParam.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public VoidResult doExecute(final NiFiClient client, final Properties properties
             throw new IllegalArgumentException("A parameter value is required when creating a new parameter");
         }
 
-        if (existingParam.isPresent() && existingParam.get().getValue().equals(paramValue)) {
```
Comment (markap14) [inline nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/params/SetParam.java:91]: This will address the case of a null value getting set to a non-null value.
But what about the opposite? If you wanted to set a populated value to a null? We'd get a NPE in that case, no?
Recommend using `Objects.equals` to check equality instead. I.e., `Objects.equals(existingParam.get().getValue(), paramValue)`

Comment (nandorsoma) [reply nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/params/SetParam.java:91]: omg yes, totally forgot about that use-case

Comment (markap14) [reply nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/params/SetParam.java:91]: Unclear to me at this point if the CLI supports setting a null value, but it probably should. If not, we may well add that in at some point, so makes sense to go ahead and support that at this point, anyway.

Comment (nandorsoma) [reply nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/params/SetParam.java:91]: So I tested it. We can avoid the null pointer exception with your proposal, but it is not enough to set null to a value. On the ParameterDTO, there is a valueRemoved flag that we also need to set true. Then the value will be updated to null on the NiFi side.
Though there is a check that only allows to update a value when the new value is not null. I don't know that it was a requirement or it is another bug: https://github.com/apache/nifi/blob/5b1034072b207a844ecbd2e8f0da8bf7981a9d24/nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/params/SetParam.java#L100

To summarize, without my change:
- CLI throws an error when we are trying to create a new Parameter with null value. On the other hand, through NiFi UI, it is possible.
- CLI silently fails to update the Parameter to null, which is also possible through NiFi UI.

Do you want me to solve these problems along with the original problem, or do you want to create a separate ticket for that? I'm happy to do them in this one, but at least we shouldn't silently fail.

Comment (markap14) [reply nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/params/SetParam.java:91]: Yeah sorry, @nandorsoma , I didn't mean to imply that settings to null via CLI should work now, or that we should support it in this Jira. Just that `Objects.equals` makes the most sense, in order to consider the possibility of either value being null :) I don't think we need to address the issue of issuing null's in this PR.

Comment (nandorsoma) [reply nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/params/SetParam.java:91]: Kk, then I just updated with your proposal @markap14. Thank you!

**Comment by markap14** [comment]
Thanks @nandorsoma merged to main
